rust/lance/src/dataset/cleanup.rs

```diff
@@ -468,6 +469,63 @@ pub async fn cleanup_old_versions(
     cleanup.run().await
 }
 
+/// If the dataset config has `lance.auto_cleanup` parameters set,
+/// this function automatically calls `dataset.cleanup_old_versions`
+/// every `lance.auto_cleanup.interval` versions. This function calls
+/// `dataset.cleanup_old_versions` with `lance.auto_cleanup.older_than`
+/// for `older_than` and `Some(false)` for both `delete_unverified` and
+/// `error_if_tagged_old_versions`.
+pub async fn auto_cleanup_hook(
+    dataset: &Dataset,
+    manifest: &Manifest,
+) -> Result<Option<RemovalStats>> {
+    if let Some(older_than) = manifest.config.get("lance.auto_cleanup.older_than") {
+        if let Some(interval) = manifest.config.get("lance.auto_cleanup.interval") {
+            let std_older_than = match parse_duration(older_than) {
+                Ok(t) => t,
+                Err(e) => {
+                    return Err(Error::Cleanup {
+                        message: format!(
+                        "Error encountered while parsing lance.auto_cleanup.older_than as std::time::Duration: {}",
+                        e
+                    ),
+                    })
+                }
+            };
+            let older_than = match TimeDelta::from_std(std_older_than) {
+                Ok(t) => t,
+                Err(e) => {
+                    return Err(Error::Cleanup {
+                        message: format!(
+                        "Error encountered while converting lance.auto_cleanup.older_than to chrono::TimeDelta: {}",
+                        e
+                    ),
+                    })
+                }
+            };
```

Comment:
Only possible reason this could fail is if it's larger than the max value, which is really high. I think we can just do
```suggestion
            let older_than = TimeDelta::from_std(std_older_than).ok_or(TimeDelta::MAX);
```

Reply:
https://github.com/lancedb/lance/pull/3572/commits/6ba14dcc653be8ae43c922d73329500c689a6e44 replaces `ok_or` with `unwrap_or`.